srcalc/src/main/java/gov/va/med/srcalc/util/csv/CsvReader.java

```diff
@@ -0,0 +1,122 @@
+package gov.va.med.srcalc.util.csv;
+
+import gov.va.med.srcalc.util.ValidationCodes;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Optional;
+
+/**
+ * General-purpose class to parse a CSV file into an object per row.
+ * @param <T> the type of object that represents each row
+ */
+public class CsvReader<T>
+{
+    private static final Logger fLogger = LoggerFactory.getLogger(CsvReader.class);
+    
+    private final RowTranslator<T> fRowTranslator;
+    
+    public CsvReader(final RowTranslator<T> rowTranslator)
+    {
+        fRowTranslator = rowTranslator;
+    }
+
+    /**
+     * Parses the given CSVRecords from a CSV file into a List of objects.
+     * @param records must be non-empty
+     * @param errors for recording validation errors
+     */
+    private TabularParseResult<T> readProcedures(
+            final List<CSVRecord> records, final Collection<TabularUploadError> errors)
+    {
+        int firstRow = 0;
+        
+        // Detect if the first row is a header and strip if so.
+        if (fRowTranslator.isHeaderRow(records.get(0)))
+        {
+            fLogger.debug("Ignoring a header row in the CSV.");
+            firstRow = 1;
+        }
+
+        // Parse each line and record errors.
```

Comment:
The comment placement is weird. I think it was supposed to be directly above the for loop.